lib/controllers/share.js

```diff
@@ -148,6 +148,14 @@ module.exports = function(hostName, port, options) {
         limit: options.shareMaxRequestSize || '200kb'
     }));
 
+    router.use(function(error, req, res, next) {
```

Comment:
Looks good. It might be good to leave comment here on why we need this handler here (to pass down 413 error returned by body-parser to the client)